web/src/components/storage/ZFCPPage.jsx

```diff
@@ -0,0 +1,726 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+// cspell:ignore wwpns npiv
+
+import React, { useCallback, useEffect, useReducer, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button, Skeleton, Toolbar, ToolbarContent, ToolbarItem } from "@patternfly/react-core";
+import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import { MainActions } from "~/components/layout";
+import { If, Page, Popup, RowActions, Section, SectionSkeleton } from "~/components/core";
+import { ZFCPDiskForm } from "~/components/storage";
+import { noop, useCancellablePromise } from "~/utils";
+import { useInstallerClient } from "~/context/installer";
+
+/**
+ * @typedef {import(~/clients/storage).ZFCPManager} ZFCPManager
+ * @typedef {import(~/clients/storage).ZFCPController} Controller
+ * @typedef {import(~/clients/storage).ZFCPDisk} Disk
+ *
+ * @typedef {object} LUN
+ * @property {string} channel
+ * @property {string} wwpn
+ * @property {string} lun
+ */
+
+/**
+ * Internal class for managing zFCP data in a format that is useful for components.
+ */
+class Manager {
```

Comment:
np: "Manager" sounds too generic here. And it is not related to `ZFCPManager` at all. As it is an internal name, it is good enough.

Reply:
Yes, I did not find a better name :/. Maybe *DataManager* ?

Reply:
Due to is an internal detail, I would keep *Manager*.